### ISSUE TITLE
context handling

### ISSUE THREAD

**Issue by sylvainlb** [nicola/simplerdf#56]
Thanks for that great work!
I was trying to use simplerdf as a replacement of the library we've been developing ourselves, also based on rdf-ext.
I'm having issues with the context though:
- context cannot be passed as a url
- `@vocab` is accepted, but not taken into account
- `@reverse` raises an error
- prefixes aren't taken into account, ie: `http://www.w3.org/ns/ldp#contains` isn't turned into `ldp:contains`

These were all working with rdf-ext.

Also, `p['@id']` isn't working. I have to do `p._core.iri.nominalValue` to get my id.

I'm not sure if I'm doing things wrong or if it is a current limitation of the library.

**Comment by bergos** [comment]
We plan to make SimpleRDF more JSON-LD compliant, but some features can't be implemented right now:
- context URLs: We haven't discussed this. We can't add this feature into the SimpleRDF class without changing the API, because it requires an async call. I would like to avoid this. But I think we could create a util, which fetches the context and returns it with a Promise API.
- `@vocab` and prefixes: `Object.defineProperty` is used to attach the properties to the SimpleRDF object. We can'T define wildcard properties. In the future we could use `Proxy`. That should be easy to implement, but that feature can't be added with a polyfill, so we have to wait till all/most engines support it.
- `@reverse`: There is already an [issue](https://github.com/simplerdf/simplerdf/issues/18) for it, but we didn't have time to implement it.
- `@id`: I created an [issue](https://github.com/simplerdf/simplerdf/issues/58) for it. Till we have this feature, you should use `.iri()`. That's just a shortcut to the property you mentioned.

**Comment by sylvainlb** [comment]
Ok, thanks. I'm not sure to understand the property thing. @vocab could be used on the graph() call, couldn't it?
I'm quite puzzled actually. Rdf-ext was handling all of this perfectly. Why didn't you use its feature?

As for the proxies, I didn't know babel couldn't cover them. But It looks like all the latest browser handle it now, so we should be able to use them soon.

**Comment by bergos** [comment]
If the JSON-LD parser and serializer would be used, any changes to/from the object from/to the graph would require an asnyc call. Also there is a big overhead and it's not possible to attach methods, because all nodes are regenerated.

Yes, I moved the define property code already to a separate file in the last PR. This refactoring was done for Proxy support. I created an [issue](https://github.com/simplerdf/simplerdf/issues/61) for it, where we can discuss further details.
